docs/contributor_guide.rst

```diff
@@ -1,18 +1,22 @@
 .. _contributor-guide:
 
-Developer documentation
-=======================
+Contributor guides
+==================
```

Comment:
If we're changing this title here we should probably change it on the card at  https://github.com/google/jax/blob/main/docs/index.rst as well